python/tests/core/constraints/test_metric_constraints.py

```diff
@@ -73,7 +73,7 @@ def metric_resolver(profile_view) -> List[Metric]:
 
     legs_less_than_12_constraint = MetricConstraint(
         name="legs less than 12",
-        condition=lambda x: x.max < 12,
+        condition=lambda x: not x.max >= 12,
```

Comment:
Make the condition less surprising when x.max is nan, because: 
```nan < 12``` evaluates to False

Reply:
Why shouldn't `NaN < 12` be false?

Reply:
or NaN is an unset value in this case. The NaN here is returned by the kll datasketch:
https://github.com/apache/datasketches-cpp/blob/dfd94962367207a13870e6b3bca5e6ed5a5ca3c6/python/src/kll_wrapper.cpp#L171-L172

I think we could change the DistributionMetric to return something else in this case rather than just passing back what we get from kll sketch, but the condition update was just incidental as the tests now cover empty distribution metrics.

The "not more than 12" sounds like a reasonable way of phrasing the condition?

Reply:
I'm only a pretend numerical analyst, but using NaN for that frightens me :)  Maybe `None` would be a little better?

The `not`s are horrible :) but we don't have to hold up this PR...

Reply:
I mean its an arbitrary condition here for the test, we could alternatively update the expected constraint results or test in other ways, I'm *not* saying you have to use the negation pattern, but this condition is valid.

We should handle these kinds of things (NaN and None aware comparisons against metrics) in our constraint helper methods. I'm not saying we should stop here but that its out of scope for the Decimals fix.